Sources/Testing/Running/Configuration.TestFilter.swift

```diff
@@ -422,3 +485,60 @@ extension Configuration.TestFilter {
     self = combining(with: other, using: op)
   }
 }
+
+// MARK: - Filterable types
+
+extension Configuration.TestFilter.Kind {
```

Comment:
Are the changes in this file actually related to suite synthesis?

Reply:
I've expanded the scope of this PR to more than just the bug fix. I'm going to rewrite the PR description to summarize, although if you think it would help, I could tease this apart into separate PRs, and just use this one to focus on the original bug fix (related to test suite synthesis).

Reply:
Yes, I do think these unrelated changes belong in their own PR. Unless they really are related, in which case please elaborate.

Reply:
I've split this out into #820